### PR TITLE
Add graphic media self label

### PR DESCRIPTION
**Pull request by haileyok**
## Why

Adding the ability to self-label as graphic media.

There's some style questions here that are not really up my alley, open to any suggestions on how to make this thing look nice. Presumably we may add more labels alongside `Graphic` in the future, so that is something to keep in mind.

Functionally though, this should work.

## Test Plan

https://github.com/user-attachments/assets/161dd486-3254-4a68-877b-4d3accbd1648

https://github.com/user-attachments/assets/bac157a9-50ed-4052-93a9-a7529d44b3da

### CODE REVIEW

**Comment by surfdude29** [comment]
This is a welcome addition to self-labelling, thank you 🙏

Sorry for asking this here, as I realise it would be out-of-scope for this PR, but I was wondering if consideration is being given to increasing the visibility and discoverability of the self-label dialog, to encourage users to add content warnings where they're appropriate, as suggested in #5383 and #5523?

**Comment by qazmlp** [comment]
That should not be mutually exclusive with adult content labels.
(Erotic content that artistically depicts mutilation is not that uncommon.)

**Comment by haileyok** [comment]
@qazmlp yup, done

**Comment by haileyok** [comment]
@surfdude29 time dependent but yea, agree.

**Comment by surfdude29** [comment]
Just testing this out on the preview branch, it's looking good! 👍

One slight issue I noticed: if a post has both `graphic-media` and one of the adult content labels applied, only the Graphic Media description is shown in the pop-up:

IMG_7415

If it's possible to show the descriptions for both labels that would be ideal, but if not it's probably not much of a problem.

**Comment by haileyok** [comment]
If I have the time yea I'll adjust that in a separate PR.

File: src/components/moderation/ContentHider.tsx [hunk 1]
```diff
@@ -34,10 +37,68 @@ export function ContentHider({
...
+      .slice(0, 2)
+      .map(cause => {
+        if (cause.type !== 'label') {
+          return
+        }
+
+        const def = cause.labelDef || getDefinition(labelDefs, cause.label)
+        if (def.identifier === 'porn' || def.identifier === 'sexual') {
+          return _(msg`Adult Content`)
+        }
```
[the remark this answers is not in the document]
Comment (haileyok) [reply src/components/moderation/ContentHider.tsx:84]: Discussed in slack. We can revisit this though for sure.

File: src/components/moderation/ContentHider.tsx [hunk 2]
```diff
@@ -34,10 +37,68 @@ export function ContentHider({
...
+        if (cause.source.type !== 'user') {
+          return false
+        }
+        if (ADULT_CONTENT_LABELS.includes(cause.label.val)) {
+          if (hasAdultContentLabel) {
+            return false
+          }
+          hasAdultContentLabel = true
+        }
```
[the remark this answers is not in the document]
Comment (haileyok) [reply src/components/moderation/ContentHider.tsx:72]: This isn't necessarily to dedupe, but rather that we only want to allow a single adult content label to show up here. The compose UI doesn't allow for it, so we don't want someone to do it in the API and sneak them into the feed UI either

**Review by estrattonbailey — APPROVED** [review]
Nice one!

**Pull request merged**